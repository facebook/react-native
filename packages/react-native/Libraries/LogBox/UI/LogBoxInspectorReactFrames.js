/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

import type LogBoxLog from '../Data/LogBoxLog';

import View from '../../Components/View/View';
import openFileInEditor from '../../Core/Devtools/openFileInEditor';
import StyleSheet from '../../StyleSheet/StyleSheet';
import Text from '../../Text/Text';
import Platform from '../../Utilities/Platform';
import LogBoxButton from './LogBoxButton';
import LogBoxInspectorSection from './LogBoxInspectorSection';
import * as LogBoxStyle from './LogBoxStyle';
import * as React from 'react';

type Props = $ReadOnly<{
  log: LogBoxLog,
}>;

const BEFORE_SLASH_RE = /^(.*)[\\/]/;

// Taken from React https://github.com/facebook/react/blob/206d61f72214e8ae5b935f0bf8628491cb7f0797/packages/react-devtools-shared/src/backend/describeComponentFrame.js#L27-L41
function getPrettyFileName(path: string) {
  let fileName = path.replace(BEFORE_SLASH_RE, '');

  // In DEV, include code for a common special case:
  // prefer "folder/index.js" instead of just "index.js".
  if (/^index\./.test(fileName)) {
    const match = path.match(BEFORE_SLASH_RE);
    if (match) {
      const pathBeforeSlash = match[1];
      if (pathBeforeSlash) {
        const folderName = pathBeforeSlash.replace(BEFORE_SLASH_RE, '');
        // Note the below string contains a zero width space after the "/" character.
        // This is to prevent browsers like Chrome from formatting the file name as a link.
        // (Since this is a source link, it would not work to open the source file anyway.)
        fileName = folderName + '/â€‹' + fileName;
      }
    }
  }

  return fileName;
}
function LogBoxInspectorReactFrames(props: Props): React.Node {
  const [collapsed, setCollapsed] = React.useState(true);
  if (
    props.log.getAvailableComponentStack() == null ||
    props.log.getAvailableComponentStack().length < 1
  ) {
    return null;
  }

  function getStackList() {
    if (collapsed) {
      return props.log.getAvailableComponentStack().slice(0, 3);
    } else {
      return props.log.getAvailableComponentStack();
    }
  }

  function getCollapseMessage() {
    if (props.log.getAvailableComponentStack().length <= 3) {
      return;
    }

    const count = props.log.getAvailableComponentStack().length - 3;
    if (collapsed) {
      return `See ${count} more components`;
    } else {
      return `Collapse ${count} components`;
    }
  }

  return (
    <LogBoxInspectorSection heading="Component Stack">
      {getStackList().map((frame, index) => (
        <View
          // Unfortunately we don't have a unique identifier for stack traces.
          key={index}
          style={componentStyles.frameContainer}>
          <LogBoxButton
            backgroundColor={{
              default: 'transparent',
              pressed: LogBoxStyle.getBackgroundColor(1),
            }}
            onPress={
              // Older versions of DevTools do not provide full path.
              // This will not work on Windows, remove check once the
              // DevTools return the full file path.
              frame.fileName.startsWith('/')
                ? () =>
                    openFileInEditor(frame.fileName, frame.location?.row ?? 1)
                : null
            }
            style={componentStyles.frame}>
            <View style={componentStyles.component}>
              <Text style={componentStyles.frameName}>
                <Text style={componentStyles.bracket}>{'<'}</Text>
                {frame.content}
                <Text style={componentStyles.bracket}>{' />'}</Text>
              </Text>
            </View>
            <Text style={componentStyles.frameLocation}>
              {getPrettyFileName(frame.fileName)}
              {frame.location ? `:${frame.location.row}` : ''}
            </Text>
          </LogBoxButton>
        </View>
      ))}
      <View style={componentStyles.collapseContainer}>
        <LogBoxButton
          backgroundColor={{
            default: 'transparent',
            pressed: LogBoxStyle.getBackgroundColor(1),
          }}
          onPress={() => setCollapsed(!collapsed)}
          style={componentStyles.collapseButton}>
          <Text style={componentStyles.collapse}>{getCollapseMessage()}</Text>
        </LogBoxButton>
      </View>
    </LogBoxInspectorSection>
  );
}

const componentStyles = StyleSheet.create({
  collapseContainer: {
    marginLeft: 15,
    flexDirection: 'row',
  },
  collapseButton: {
    borderRadius: 5,
  },
  collapse: {
    color: LogBoxStyle.getTextColor(0.7),
    fontSize: 12,
    fontWeight: '300',
    lineHeight: 20,
    marginTop: 0,
    paddingVertical: 5,
    paddingHorizontal: 10,
  },
  frameContainer: {
    flexDirection: 'row',
    paddingHorizontal: 15,
  },
  frame: {
    flex: 1,
    paddingVertical: 4,
    paddingHorizontal: 10,
    borderRadius: 5,
  },
  component: {
    flexDirection: 'row',
    paddingRight: 10,
  },
  frameName: {
    fontFamily: Platform.select({android: 'monospace', ios: 'Menlo'}),
    color: LogBoxStyle.getTextColor(1),
    fontSize: 14,
    includeFontPadding: false,
    lineHeight: 18,
  },
  bracket: {
    fontFamily: Platform.select({android: 'monospace', ios: 'Menlo'}),
    color: LogBoxStyle.getTextColor(0.4),
    fontSize: 14,
    fontWeight: '500',
    includeFontPadding: false,
    lineHeight: 18,
  },
  frameLocation: {
    color: LogBoxStyle.getTextColor(0.7),
    fontSize: 12,
    fontWeight: '300',
    includeFontPadding: false,
    lineHeight: 16,
    paddingLeft: 10,
  },
});

export default LogBoxInspectorReactFrames;
