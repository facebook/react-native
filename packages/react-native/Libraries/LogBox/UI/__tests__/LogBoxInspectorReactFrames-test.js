/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall react_native
 */

'use strict';

const render = require('../../../../jest/renderer');
const LogBoxLog = require('../../Data/LogBoxLog').default;
const LogBoxInspectorReactFrames =
  require('../LogBoxInspectorReactFrames').default;
const React = require('react');

// Mock child components because we are interested in snapshotting the behavior
// of `LogBoxInspectorReactFrames`, not its children.
jest.mock('../LogBoxButton', () => ({
  __esModule: true,
  default: 'LogBoxButton',
}));
jest.mock('../LogBoxInspectorSection', () => ({
  __esModule: true,
  default: 'LogBoxInspectorSection',
}));

describe('LogBoxInspectorReactFrames', () => {
  it('should render null for no componentStack frames', async () => {
    const output = await render.create(
      <LogBoxInspectorReactFrames
        log={
          new LogBoxLog({
            level: 'warn',
            isComponentError: false,
            message: {
              content: 'Some kind of message',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message',
            componentStack: [],
          })
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render componentStack frames without full path pressable', async () => {
    const output = await render.create(
      <LogBoxInspectorReactFrames
        log={
          new LogBoxLog({
            level: 'warn',
            isComponentError: false,
            message: {
              content: 'Some kind of message',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message',
            componentStack: [
              {
                content: 'MyComponent',
                fileName: 'MyComponentFile.js',
                location: {
                  row: 1,
                  column: -1,
                },
              },
            ],
          })
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render componentStack frames with full path pressable', async () => {
    const output = await render.create(
      <LogBoxInspectorReactFrames
        log={
          new LogBoxLog({
            level: 'warn',
            isComponentError: false,
            message: {
              content: 'Some kind of message',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message',
            componentStack: [
              {
                content: 'MyComponent',
                fileName: '/path/to/MyComponentFile.js',
                location: {
                  row: 1,
                  column: -1,
                },
              },
            ],
          })
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render componentStack frames with parent folder of index.js', async () => {
    const output = await render.create(
      <LogBoxInspectorReactFrames
        log={
          new LogBoxLog({
            level: 'warn',
            isComponentError: false,
            message: {
              content: 'Some kind of message',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message',
            componentStack: [
              {
                content: 'MyComponent',
                fileName: '/path/to/index.js',
                location: {
                  row: 1,
                  column: -1,
                },
              },
            ],
          })
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render componentStack frames with more than 3 stacks', async () => {
    const output = await render.create(
      <LogBoxInspectorReactFrames
        log={
          new LogBoxLog({
            level: 'warn',
            isComponentError: false,
            message: {
              content: 'Some kind of message',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message',
            componentStack: [
              {
                content: 'MyComponent',
                fileName: '/path/to/index.js',
                location: {
                  row: 1,
                  column: -1,
                },
              },
              {
                content: 'MyComponent2',
                fileName: '/path/to/index2.js',
                location: {
                  row: 1,
                  column: -1,
                },
              },
              {
                content: 'MyComponent3',
                fileName: '/path/to/index3.js',
                location: {
                  row: 1,
                  column: -1,
                },
              },
              {
                content: 'MyComponent4',
                fileName: '/path/to/index4.js',
                location: {
                  row: 1,
                  column: -1,
                },
              },
            ],
          })
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });
});
