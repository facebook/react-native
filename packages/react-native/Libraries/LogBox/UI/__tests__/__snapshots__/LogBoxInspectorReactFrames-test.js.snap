// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogBoxInspectorReactFrames should render componentStack frames with full path pressable 1`] = `
<LogBoxInspectorSection
  heading="Component Stack"
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingHorizontal": 15,
      }
    }
  >
    <LogBoxButton
      backgroundColor={
        Object {
          "default": "transparent",
          "pressed": "rgba(51, 51, 51, 1)",
        }
      }
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 5,
          "flex": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 4,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingRight": 10,
          }
        }
      >
        <Text
          id="logbox_component_stack_frame_text"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Menlo",
              "fontSize": 14,
              "includeFontPadding": false,
              "lineHeight": 18,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 0.4)",
                "fontFamily": "Menlo",
                "fontSize": 14,
                "fontWeight": "500",
                "includeFontPadding": false,
                "lineHeight": 18,
              }
            }
          >
            &lt;
          </Text>
          MyComponent
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 0.4)",
                "fontFamily": "Menlo",
                "fontSize": 14,
                "fontWeight": "500",
                "includeFontPadding": false,
                "lineHeight": 18,
              }
            }
          >
             /&gt;
          </Text>
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 12,
            "fontWeight": "300",
            "includeFontPadding": false,
            "lineHeight": 16,
            "paddingLeft": 10,
          }
        }
      >
        MyComponentFile.js
        :1
      </Text>
    </LogBoxButton>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 15,
      }
    }
  >
    <LogBoxButton
      backgroundColor={
        Object {
          "default": "transparent",
          "pressed": "rgba(51, 51, 51, 1)",
        }
      }
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 12,
            "fontWeight": "300",
            "lineHeight": 20,
            "marginTop": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
      />
    </LogBoxButton>
  </View>
</LogBoxInspectorSection>
`;

exports[`LogBoxInspectorReactFrames should render componentStack frames with more than 3 stacks 1`] = `
<LogBoxInspectorSection
  heading="Component Stack"
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingHorizontal": 15,
      }
    }
  >
    <LogBoxButton
      backgroundColor={
        Object {
          "default": "transparent",
          "pressed": "rgba(51, 51, 51, 1)",
        }
      }
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 5,
          "flex": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 4,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingRight": 10,
          }
        }
      >
        <Text
          id="logbox_component_stack_frame_text"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Menlo",
              "fontSize": 14,
              "includeFontPadding": false,
              "lineHeight": 18,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 0.4)",
                "fontFamily": "Menlo",
                "fontSize": 14,
                "fontWeight": "500",
                "includeFontPadding": false,
                "lineHeight": 18,
              }
            }
          >
            &lt;
          </Text>
          MyComponent
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 0.4)",
                "fontFamily": "Menlo",
                "fontSize": 14,
                "fontWeight": "500",
                "includeFontPadding": false,
                "lineHeight": 18,
              }
            }
          >
             /&gt;
          </Text>
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 12,
            "fontWeight": "300",
            "includeFontPadding": false,
            "lineHeight": 16,
            "paddingLeft": 10,
          }
        }
      >
        to/â€‹index.js
        :1
      </Text>
    </LogBoxButton>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingHorizontal": 15,
      }
    }
  >
    <LogBoxButton
      backgroundColor={
        Object {
          "default": "transparent",
          "pressed": "rgba(51, 51, 51, 1)",
        }
      }
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 5,
          "flex": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 4,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingRight": 10,
          }
        }
      >
        <Text
          id="logbox_component_stack_frame_text"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Menlo",
              "fontSize": 14,
              "includeFontPadding": false,
              "lineHeight": 18,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 0.4)",
                "fontFamily": "Menlo",
                "fontSize": 14,
                "fontWeight": "500",
                "includeFontPadding": false,
                "lineHeight": 18,
              }
            }
          >
            &lt;
          </Text>
          MyComponent2
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 0.4)",
                "fontFamily": "Menlo",
                "fontSize": 14,
                "fontWeight": "500",
                "includeFontPadding": false,
                "lineHeight": 18,
              }
            }
          >
             /&gt;
          </Text>
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 12,
            "fontWeight": "300",
            "includeFontPadding": false,
            "lineHeight": 16,
            "paddingLeft": 10,
          }
        }
      >
        index2.js
        :1
      </Text>
    </LogBoxButton>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingHorizontal": 15,
      }
    }
  >
    <LogBoxButton
      backgroundColor={
        Object {
          "default": "transparent",
          "pressed": "rgba(51, 51, 51, 1)",
        }
      }
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 5,
          "flex": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 4,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingRight": 10,
          }
        }
      >
        <Text
          id="logbox_component_stack_frame_text"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Menlo",
              "fontSize": 14,
              "includeFontPadding": false,
              "lineHeight": 18,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 0.4)",
                "fontFamily": "Menlo",
                "fontSize": 14,
                "fontWeight": "500",
                "includeFontPadding": false,
                "lineHeight": 18,
              }
            }
          >
            &lt;
          </Text>
          MyComponent3
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 0.4)",
                "fontFamily": "Menlo",
                "fontSize": 14,
                "fontWeight": "500",
                "includeFontPadding": false,
                "lineHeight": 18,
              }
            }
          >
             /&gt;
          </Text>
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 12,
            "fontWeight": "300",
            "includeFontPadding": false,
            "lineHeight": 16,
            "paddingLeft": 10,
          }
        }
      >
        index3.js
        :1
      </Text>
    </LogBoxButton>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 15,
      }
    }
  >
    <LogBoxButton
      backgroundColor={
        Object {
          "default": "transparent",
          "pressed": "rgba(51, 51, 51, 1)",
        }
      }
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 12,
            "fontWeight": "300",
            "lineHeight": 20,
            "marginTop": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
      >
        See 1 more components
      </Text>
    </LogBoxButton>
  </View>
</LogBoxInspectorSection>
`;

exports[`LogBoxInspectorReactFrames should render componentStack frames with parent folder of index.js 1`] = `
<LogBoxInspectorSection
  heading="Component Stack"
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingHorizontal": 15,
      }
    }
  >
    <LogBoxButton
      backgroundColor={
        Object {
          "default": "transparent",
          "pressed": "rgba(51, 51, 51, 1)",
        }
      }
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 5,
          "flex": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 4,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingRight": 10,
          }
        }
      >
        <Text
          id="logbox_component_stack_frame_text"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Menlo",
              "fontSize": 14,
              "includeFontPadding": false,
              "lineHeight": 18,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 0.4)",
                "fontFamily": "Menlo",
                "fontSize": 14,
                "fontWeight": "500",
                "includeFontPadding": false,
                "lineHeight": 18,
              }
            }
          >
            &lt;
          </Text>
          MyComponent
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 0.4)",
                "fontFamily": "Menlo",
                "fontSize": 14,
                "fontWeight": "500",
                "includeFontPadding": false,
                "lineHeight": 18,
              }
            }
          >
             /&gt;
          </Text>
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 12,
            "fontWeight": "300",
            "includeFontPadding": false,
            "lineHeight": 16,
            "paddingLeft": 10,
          }
        }
      >
        to/â€‹index.js
        :1
      </Text>
    </LogBoxButton>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 15,
      }
    }
  >
    <LogBoxButton
      backgroundColor={
        Object {
          "default": "transparent",
          "pressed": "rgba(51, 51, 51, 1)",
        }
      }
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 12,
            "fontWeight": "300",
            "lineHeight": 20,
            "marginTop": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
      />
    </LogBoxButton>
  </View>
</LogBoxInspectorSection>
`;

exports[`LogBoxInspectorReactFrames should render componentStack frames without full path pressable 1`] = `
<LogBoxInspectorSection
  heading="Component Stack"
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingHorizontal": 15,
      }
    }
  >
    <LogBoxButton
      backgroundColor={
        Object {
          "default": "transparent",
          "pressed": "rgba(51, 51, 51, 1)",
        }
      }
      onPress={null}
      style={
        Object {
          "borderRadius": 5,
          "flex": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 4,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingRight": 10,
          }
        }
      >
        <Text
          id="logbox_component_stack_frame_text"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Menlo",
              "fontSize": 14,
              "includeFontPadding": false,
              "lineHeight": 18,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 0.4)",
                "fontFamily": "Menlo",
                "fontSize": 14,
                "fontWeight": "500",
                "includeFontPadding": false,
                "lineHeight": 18,
              }
            }
          >
            &lt;
          </Text>
          MyComponent
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 0.4)",
                "fontFamily": "Menlo",
                "fontSize": 14,
                "fontWeight": "500",
                "includeFontPadding": false,
                "lineHeight": 18,
              }
            }
          >
             /&gt;
          </Text>
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 12,
            "fontWeight": "300",
            "includeFontPadding": false,
            "lineHeight": 16,
            "paddingLeft": 10,
          }
        }
      >
        MyComponentFile.js
        :1
      </Text>
    </LogBoxButton>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 15,
      }
    }
  >
    <LogBoxButton
      backgroundColor={
        Object {
          "default": "transparent",
          "pressed": "rgba(51, 51, 51, 1)",
        }
      }
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 12,
            "fontWeight": "300",
            "lineHeight": 20,
            "marginTop": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
      />
    </LogBoxButton>
  </View>
</LogBoxInspectorSection>
`;

exports[`LogBoxInspectorReactFrames should render null for no componentStack frames 1`] = `null`;
